Tomlet/TomlCompositeDeserializer.cs

```diff
@@ -121,4 +114,47 @@ public static TomlSerializationMethods.Deserialize<object> For(Type type)
 
         return deserializer;
     }
+
+    private static object CreateInstance(Type type, TomlValue tomlValue, TomlSerializerOptions options, out HashSet<string> assignedMembers)
+    {
+        if (!type.TryGetBestMatchConstructor(out var constructor))
+        {
+            throw new TomlInstantiationException();
+        }
+
+        var parameters = constructor!.GetParameters();
+        if (parameters.Length == 0)
+        {
+            assignedMembers = new HashSet<string>();
+            return Activator.CreateInstance(type)!;
+        }
+
+        if (tomlValue is not TomlTable table)
```

Comment:
I would do this before TryGetBestMatchConstructor, as that method seems moderately expensive, so get the cheap checks out of the way first.